src/ngraph/event_tracing.cpp

```diff
@@ -0,0 +1,76 @@
+//*****************************************************************************
+// Copyright 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "event_tracing.hpp"
+#include "nlohmann/json.hpp"
+
+using namespace std;
+
+mutex ngraph::Event::s_file_mutex;
+ofstream ngraph::Event::s_event_log;
+bool ngraph::Event::s_tracing_enabled = false;
+
+void ngraph::Event::write_trace(const ngraph::Event& event)
+{
+    lock_guard<mutex> lock(s_file_mutex);
+    if (!is_tracing_enabled())
+    {
+        return;
+    }
+
+    static bool so_initialized = false;
+    if (!so_initialized)
+    {
+        // Open the file
+        s_event_log.open("ngraph_event_trace.json", ios_base::trunc);
```

Comment:
Would be nice to make this filename customizable somehow, perhaps through an environment variable. No need to address in this PR.

Reply:
We will address this in a future PR.